app/models/user.rb

```diff
@@ -737,4 +737,10 @@ def check_permissions_for_changing_login
       end
     end
   end
+
+  def user_disabled_itself
+    if disabled? && self == User.current
+      errors.add :disabled, _('It is not possible to disable himself')
```

Comment:
Maybe it's better to say 'It is not possible to disable yourself'? If I'd try to disable myself I'd expect that message.